### ISSUE TITLE
The $length function returns the Cartesian value when called from the aggregate function

### ISSUE THREAD

**Issue by pigreco** [qgis/QGIS#55657]
### What is the bug or the crash?

problem description:
I have a linear layer (testOne) and a table (tableOne) without geometry,
I start the field calc from the table and write the following expression to update the len attribute:

```
aggregate(
layer:='testOne',
aggregate:='array_agg',
expression:=$length) [0]
```

the result (126671.709 m) is different from what I would expect (126721.310 m), as it returns the Cartesian value of `$length` and not the ellipsoid one, in fact if I used `$length` in the testOne vector I would get the desired value.

data:
https://mega.nz/file/pFQnkYxJ#3zNJR0rzH-a4opip3rgkbJoLrRUlNu7-vwXeiUra_Hs

### Steps to reproduce the issue

1. create a linear vector with projected SR and add a single line;
2. create a simple table with a single row;
3. When updating the table, use the expression below:

```
aggregate(
layer:='testOne',
aggregate:='array_agg',
expression:=$length) [0]
```

## OSGeo4W 64b
 Tested on QGIS 3.28.13 LTR Florence and on QGIS 3.34.1 Prizren by OSGeo4W - win 10 PRO 64b

Versione di QGIS
3.28.13-Firenze
Revisione codice QGIS
3b4db4f561
Versione Qt
5.15.3
Versione Python
3.9.5
Versione GDAL/OGR
3.8.1
Versione PROJ
9.3.1
Versione database del Registro EPSG
v10.098 (2023-11-24)
Versione GEOS
3.12.1-CAPI-1.18.1
Versione SQLite
3.41.1
Versione PDAL
2.6.0
Versione client PostgreSQL
unknown
Versione SpatiaLite
5.1.0
Versione QWT
6.1.6
Versione QScintilla2
2.13.4
Versione SO
Windows 10 Version 2009

### Supported QGIS version

- [X] I'm running a supported QGIS version according to [the roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule).

### New profile

- [X] I tried with a new [QGIS profile](https://docs.qgis.org/latest/en/docs/user_manual/introduction/qgis_configuration.html#working-with-user-profiles)

### Additional context

the project is set up correctly with ellipsoid, i.e. we know that: 'The length calculated by this function respects both the current project's ellipsoid setting and the distance unit settings'

I attach data and example project.
https://mega.nz/file/pFQnkYxJ#3zNJR0rzH-a4opip3rgkbJoLrRUlNu7-vwXeiUra_Hs

**Comment by agiudiceandrea** [comment]
@pigreco it seems to me the issue no longer occurs using the latest QGIS 3.38 and QGIS 3.34. May you double check?

**Comment by pigreco** [comment]
> it seems to me the issue no longer occurs using the latest QGIS 3.38 and QGIS 3.34. May you double check?

The problem still exists if I start the field calc from the `tableOne` table and launch the expression:

```
aggregate(
layer:='testOne',
aggregate:='array_agg',
expression:=$length) [0]
```
it returns 126671.709 and not 126721.310

![image](https://github.com/user-attachments/assets/f6c723d8-4dae-45de-b028-c5781659127b)

## OSGeo4W QGIS 3.34.10 ltr WIN 11

**Comment by agiudiceandrea** [comment]
@pigreco, you are right! The issue report's title let my think the issue occurs in any circumstance, while it occurs for field calculations made on non geometry layers.
